crates/pathfinder/src/rpc/serde.rs

```diff
@@ -178,9 +290,41 @@ pub(crate) fn bytes_to_hex_str(bytes: &[u8]) -> String {
         buf[pos..pos + 2].copy_from_slice(&x);
     });
     buf[1] = b'x';
+    &buf[..len]
+}
 
+/// A convenience function which produces a "0x" prefixed hex str slice in a given buffer `buf`
+/// from an array of bytes.
+/// Panics if `bytes.len() * 2 + 2 > buf.len()`
+pub(crate) fn bytes_as_hex_str<'a>(bytes: &'a [u8], buf: &'a mut [u8]) -> &'a str {
```

Comment:
This would probably work without calling the first argument `'a` (`'_` if need be, but can't see why it wouldn't work without it), at least it would be more clear, since we only really want to hold on to the second argument.

```suggestion
pub(crate) fn bytes_as_hex_str<'a>(bytes: &[u8], buf: &'a mut [u8]) -> &'a str {
```

A similar applies to `it_to_hex_string` where iterator needs to return `&'a u8` but they don't need to outlive the function or be constrained by the return value. I am surprised the iterator did not need an implicit `impl Iterator<...> + '_` but perhaps that's the only possibility to call this fn.

Reply:
Lemme check it :+1: .